binderhub/main.py

```diff
@@ -35,6 +35,7 @@ def get(self):
             google_analytics_code=self.settings['google_analytics_code'],
             google_analytics_domain=self.settings['google_analytics_domain'],
             extra_footer_scripts=self.settings['extra_footer_scripts'],
+            repo_providers=SPEC_NAMES,
```

Comment:
I think we should at least filter it by `repo_providers` config:
```python
            repo_providers={prefix: SPEC_NAMES[prefix] for prefix in SPEC_NAMES if prefix in self.settings['repo_providers']},
```
But probably better to use `self.settings['repo_providers']` itself directly:

```python
            repo_providers=self.settings['repo_providers'],
```
Together with this we can add `display_name` attribute in repo provider classes and use it in dropdown menu.